Lib/test/test_configparser.py

```diff
@@ -720,6 +721,11 @@ def test_read_returns_file_list(self):
         parsed_files = cf.read(file1)
         self.assertEqual(parsed_files, [file1])
         self.assertEqual(cf.get("Foo Bar", "foo"), "newbar")
+        # check when we pass only a Path object:
+        cf = self.newconfig()
+        parsed_files = cf.read(pathlib.Path(file1))
+        self.assertEqual(parsed_files, [pathlib.Path(file1)])
```

Comment:
> I'm not certain it makes as much sense here as it does in zipfiles as the config file doesn't contain the same kind of internal structure, are there other examples where it's being converted?

The only other example where the file path is part of the public API is gzip module and it's converted to `str` by using `os.fspath()`: https://hg.python.org/cpython/rev/b244bf74b638